aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -242,6 +272,10 @@ impl<T: Hash + Clone + Debug + Eq + Serialize, V: TransactionWrite> VersionedGro
             );
         }
 
+        if !replacing && !idx_update_tags.is_empty() {
```

Comment:
I think this is a bit confusing that the caller `write` changes `self.size_changed` outside of this function, but `pub remove` relies on internal changes. Maybe it is better to not have this check here so it is used by write without passing the flag... and have a wrapper remove function here which would be called by `pub remove` and do the extra flag setting?

Reply:
flag was an improvement from the previous iteration and got stuck there - this is even better, thanks!